docs/proposals/Certificate Revocation Lists.md

```diff
@@ -0,0 +1,188 @@
+# CRL and CRL Cache Design
+
+## Intro
+
+Certificate validation is an essential step during signature validation. Currently Ratify supports checking for revoked certificates through OCSP supported by notation-go library. However, OCSP validation requires internet connection for each validation while CRL could be cached for better performance. As notary-project is adding the CRL support for notation signature validation, Ratify could utilize it.
+
+## Goals
+
+CRL support, including CRL downloading, validation, and revocation list checks.
+
+- Define a cache provider interface for CRL
+- Implement default file-based cache implementation for both CLI and K8S
+- Implement preload CRL when cert added from KMP
+- Update CRL and CRL caching related documentation
+
+## Design Points
+
+**How to Get CRL**
+
+With no extra configuration, CRL download location (URL) can be obtained from the certificate's CRL Distribution Point (CDP) extension. If the CRL cannot be downloaded within the timeout threshold the revocation result will be "revocation unavailable". More details are showing in the Download CRL section
```

Comment:
Hi @junczhu , will you be able to add an example one provider? eg. AKV senario?

Reply:
Sure.

Reply:
[microsoft.com/pkiops/docs/repository.htm?msockid=2f918efdcbb362cf2f319a35cabb634f](https://www.microsoft.com/pkiops/docs/repository.htm?msockid=2f918efdcbb362cf2f319a35cabb634f)